mpt_trie/src/partial_trie.rs

```diff
@@ -329,6 +356,18 @@ impl PartialTrie for HashedPartialTrie {
         Self {
             node,
             hash: Arc::new(RwLock::new(None)),
+            orphaned_hash_node_strategy: OnOrphanedHashNode::default(),
+        }
+    }
+
+    fn new_with_strategy(
+        node: Node<Self>,
+        orphaned_hash_node_strategy: OnOrphanedHashNode,
```

Comment:
nit: long

Reply:
Do you find the method argument long, or the struct field long? I liked `collapse_strategy` but renamed it out of consistency after tweaking the terminology.

Reply:
I think both are long - the type itself is descriptive enough, but I don't think it's a big deal :)